Revise PCMDI-AMIP-* and prototype data; catchup on DOI assignments (#318)
PCMDI/input4MIPs_CVs: changelog/317.trivial.md

> updating PCMDI-AMIP-* prototype datasets in preparation for CMIP7Plus mip_era

Reviewer: ```suggestion
- Updated PCMDI-AMIP-* prototype datasets in preparation for CMIP7Plus mip_era
```

> catchup on some previous OSTI DOI assignments for published datasets

Reviewer: ```suggestion
- Added DOI information for some published datasets which were missing it
```

I'd also `mv changelog/317.trivial.md` to `changelog/317.docs.md` I think. Users might be interested in this change and if it's under trivial it's not shown when we we release
